src/test/java/com/neo4j/docker/TestDockerComposeSecrets.java

```diff
@@ -0,0 +1,113 @@
+                                       dockerComposeContainer.getServicePort( serviceName, DEFAULT_BOLT_PORT ) );
+            dbio.verifyConnectivity( "neo4j", "simplecontainerpassword" );
+        }
+    }
+
+    @Test
+    void shouldCreateContainerWithSecretPasswordAndConnect() throws Exception
+    {
+        var tmpDir = temporaryFolderManager.createFolder( "Container_Compose_With_Secrets" );
+        var composeFile = copyDockerComposeResourceFile( tmpDir, "container-compose-with-secrets.yml" );
+        var serviceName = "secretscontainer";
+
+        var newSecretPassword = "neo4j/newSecretPassword";
+        Files.createFile( tmpDir.resolve( "neo4j_auth.txt" ) );
+        Files.writeString( tmpDir.resolve( "neo4j_auth.txt" ), newSecretPassword );
+
+        try ( var dockerComposeContainer = createContainer( composeFile, tmpDir, serviceName ) )
+        {
+            dockerComposeContainer.start();
+            var dbio = new DatabaseIO( dockerComposeContainer.getServiceHost( serviceName, DEFAULT_BOLT_PORT ),
+                                       dockerComposeContainer.getServicePort( serviceName, DEFAULT_BOLT_PORT ) );
+            dbio.verifyConnectivity( "neo4j", "newSecretPassword" );
+        }
+    }
+
+    @Test
+    void shouldOverrideVariableWithSecretValue() throws Exception
+    {
+        var tmpDir = temporaryFolderManager.createFolder( "Container_Compose_With_Secrets_Override" );
+        var composeFile = copyDockerComposeResourceFile( tmpDir, "container-compose-with-secrets-override.yml" );
+        var serviceName = "secretsoverridecontainer";
+
+        var newSecretPageCache = "50M";
+        Files.createFile( tmpDir.resolve( "neo4j_pagecache.txt" ) );
+        Files.writeString( tmpDir.resolve( "neo4j_pagecache.txt" ), newSecretPageCache );
+
+        try ( var dockerComposeContainer = createContainer( composeFile, tmpDir, serviceName ) )
+        {
+            dockerComposeContainer.start();
+
+            var configFile = tmpDir.resolve( "neo4j" ).resolve( "config" ).resolve( "neo4j.conf" ).toFile();
+            Assertions.assertTrue( configFile.exists(), "neo4j.conf file does not exist" );
+            Assertions.assertTrue( configFile.canRead(), "cannot read neo4j.conf file" );
+
+            Assertions.assertFalse( Files.readAllLines( configFile.toPath() ).contains( "dbms.memory.pagecache.size=10M" ) );
+            Assertions.assertTrue( Files.readAllLines( configFile.toPath() ).contains( "dbms.memory.pagecache.size=50M" ) );
+        }
+    }
+
+    private File copyDockerComposeResourceFile( Path targetDirectory, String fileName ) throws IOException
+    {
+        File compose_file = new File( targetDirectory.toString(), fileName );
+        if ( compose_file.exists() )
+        {
+            Files.delete( compose_file.toPath() );
+        }
+        Files.copy( Objects.requireNonNull( getClass().getClassLoader().getResourceAsStream( fileName ) ), Paths.get( compose_file.getPath() ) );
```

Comment:
I think there are enough compose files that we should put them in a subfolder of `src/test/resources`
[TestPluginInstallation.java](https://github.com/neo4j/docker-neo4j/blob/17dd78c568d34b079f6de3837e926fa89c6aae11/src/test/java/com/neo4j/docker/coredb/plugins/TestPluginInstallation.java#L142) has an example of how to retrieve files in subfolders of test resources. [Configuration.java](https://github.com/neo4j/docker-neo4j/blob/17dd78c568d34b079f6de3837e926fa89c6aae11/src/test/java/com/neo4j/docker/coredb/configurations/Configuration.java#L85) has a different method of getting them.